### PR TITLE
MISP2021 E2E ASR Baseline

### PR DESCRIPTION
**Pull request by neillu23**

### CODE REVIEW

File: egs2/misp2021/asr1/local/enhancement.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# use nara-wpe and BeamformIt to enhance multichannel data
+
+set -e -o pipefail
+
+# configs
+stage=0
+nj=6
+cmd=run.pl
+
+
```
Comment (ftshijt) [inline egs2/misp2021/asr1/local/enhancement.sh:14]: Can you add `.utils/parse_options.sh` here to parse the options from command line

File: egs2/misp2021/asr1/run.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+use_word_lm=false
+
+
+./asr.sh                                   \
+    --lang zh \
+    --audio_format wav                     \
+    --nlsyms_txt data/nlsyms.txt           \
+    --ngpu 3                               \
+    --token_type char                      \
+    --feats_type fbank_pitch               \
```
Comment (ftshijt) [inline egs2/misp2021/asr1/run.sh:27]: is it possible that we stick to `raw` features for espnet2 (it's ok to use fbank_pitch, I'm just curious as we do not want to have dependencies on Kaldi for espnet2)

File: egs2/misp2021/asr1/local/run_wpe.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+# _*_ coding: UTF-8 _*_
+import os
+import codecs
+import argparse
+import numpy as np
+from tqdm import tqdm
+import scipy.io.wavfile as wf
+from multiprocessing import Pool
+from nara_wpe.wpe import wpe_v8 as wpe
+from nara_wpe.utils import stft, istft
```
Comment (ftshijt) [inline egs2/misp2021/asr1/local/run_wpe.py:11]: Do we have to add additional requirements for nara-wpe as a package?

**Review by ftshijt — COMMENTED** [review]
Very cool, just some minor comments

File: egs2/misp2021/asr1/local/run_wpe.py (hunk 3 above)
Comment (neillu23) [reply egs2/misp2021/asr1/local/run_wpe.py:11]: Is the detection and download process for the nara_wpe package in local/enhancement.sh enough for handling this? 
`pip show -f nara_wpe >/dev/null || pip install nara_wpe`

Comment (ftshijt) [reply egs2/misp2021/asr1/local/run_wpe.py:11]: Looks good! thx

**Comment by ftshijt** [comment]
LGTM! thx

**Comment by sw005320** [comment]
@neillu23, can you also add a pre-trained model?

**Pull request merged**